### ISSUE TITLE
Enum toString not working with React frontend 

### ISSUE THREAD

**Issue by eeriksp** [jhipster/generator-jhipster#9773]
##### **Overview of the issue**

If I open the form for adding a new entity, dropdown fields representing enums do not show a correct string representation as seen in the picture. The browser console does not show any errors.

However, if a view the list page showing all the entity objects, all enumerations have a correct string representation.
![image](https://user-images.githubusercontent.com/35540617/58080759-73e58300-7bbc-11e9-9def-4cf1e5382986.png)

##### **JHipster Version(s)**

```
cleverbin@0.0.0 C:\development\CleverBin\backend
+-- generator-jhipster@6.0.1
`-- generator-jhipster-kotlin@0.8.0
  `-- generator-jhipster@6.0.1  deduped

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster-kotlin": {
    "promptValues": {
      "packageName": "ee.smartworks.cleverbin"
    },
    "jhipsterVersion": "6.0.1",
    "applicationType": "monolith",
    "baseName": "cleverbin",
    "packageName": "ee.smartworks.cleverbin",
    "packageFolder": "ee/smartworks/cleverbin",
    "serverPort": "8080",
    "authenticationType": "session",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "gradle",
    "enableSwaggerCodegen": false,
    "rememberMeKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons"
  },
  "generator-jhipster": {
    "promptValues": {
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.0.1",
    "applicationType": "monolith",
    "baseName": "cleverbin",
    "clientFramework": "react",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [
      {
        "name": "generator-jhipster-kotlin",
        "version": "0.8.0"
      }
    ],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en", "et"],
    "blueprint": "generator-jhipster-kotlin",
    "blueprintVersion": "0.8.0"
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Location {
  address String required
}
entity Bin {
  state BinState required,
  garbageLevel GarbageLevel required
}
entity BinSession {
  timestamp Instant required,
  status BinSessionStatus required,
  garbageWeight Integer
}

enum BinState {
  FREE,
  BUSY,
  ERROR
}

enum GarbageLevel {
  NOT_FULL,
  ALMOST_FULL,
  FULL
}

enum BinSessionStatus {
  STARTED,
  BIN_ACTIVATED,
  FINISHED,
  FAILED
}

relationship OneToMany {
  Location{bins} to Bin{location required},
  Bin{binSessions} to BinSession{bin required}
}
relationship ManyToOne {
  BinSession{user required} to User
}

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)

git version 2.14.1.windows.1

node: v10.15.3

npm: 6.9.0

yeoman: 2.0.6
##### **Browsers and Operating System**

Using Windows 10 and Google Chrome

-   [ x ] Checking this box is mandatory (this is just to show you read everything)

**Comment by clement26695** [comment]
I think the issue is now resolved on master (#9730), but is not in the latest release yet.

**Comment by ruddell** [comment]
Tested the JDL and the enums are displayed correctly with the master branch.
